cql3/type_json.cc

```diff
@@ -75,6 +77,31 @@ static sstring quote_json_string(const sstring& value) {
     return oss.str();
 }
 
+template <std::floating_point Tf, std::integral Ti>
+static std::partial_ordering cmp(Tf x, Ti y) {
+    return x <=> y;
+}
+
+// we are not able to fit an uint64_t or a 64 bit float into a wider numeric type
+// to compare them, so need to do this in a more convoluted way. specifically, we
+// have to compare the float with the maximum value represented by the integer
+// type to see if the nearest conversion changes the result of the comparison.
+//
+// see https://github.com/JuliaLang/julia/blob/49b361aa2ec68efe4cb0d5870e772eb3e0d89993/base/float.jl#L553-L565
+template <std::floating_point Tf, std::integral Ti>
+requires (sizeof(Tf) <= 8 && sizeof(Ti) >= 8)
```

Comment:
What in the code below requires size 8? Isn't just sizeof(Tf) <= sizeof(Ti) enough?

Reply:
because 64bit is the problem i am trying to fix. if the LHS is a 32-bit float point number, and the RHS is a 32-bit integer, we can just cast them to a 64 bit float point numbers and compare them.

Reply:
Does this code actually work for things which aren't specifically double and int_64_t?
E.g., if x is 32-bit (float) and y is 128-bit integer, does this code actually work? I'm not sure I understand why.  It seems like it checks "x < fy" but the 128-bit integer doesn't fit the floating point so... I don't understand what the code does. Again, a unit test might have convinced me it's correct.

Reply:
float32 versus. int128 should work. if this is something we support, will surely add tests for it. was just reluctant  to test an internal helper.. even if we will make a public one for testing it.